qt/scientific_interfaces/Direct/test/ALFAnalysisPresenterTest.h

```diff
@@ -205,8 +205,25 @@ class ALFAnalysisPresenterTest : public CxxTest::TestSuite {
     m_presenter->notifyExportWorkspaceToADSClicked();
   }
 
-  void test_that_notifyExternalPlotClicked_calls_the_expected_model_function() {
-    EXPECT_CALL(*m_model, openExternalPlot()).Times(1);
+  void test_that_notifyExternalPlotClicked_will_open_an_external_plot_from_view() {
+    auto const workspace = WorkspaceCreationHelper::create2DWorkspace(1, 100);
+    std::vector<int> const workspaceIndices{0, 1};
+
+    EXPECT_CALL(*m_model, plottedWorkspace()).Times(1).WillOnce(Return(workspace));
+
+    EXPECT_CALL(*m_model, plottedWorkspaceIndices()).Times(1).WillOnce(Return(workspaceIndices));
+    EXPECT_CALL(*m_view, openExternalPlot(_, workspaceIndices)).Times(1);
```

Comment:
Can we pass the workspace through here too? It's sometimes awkward dealing with the pointers but should be possible - let me know if you have problems. I often find that wrapping it in the `Eq` matcher explicity i.e. `Eq(workspace)` can resolve problems.